src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -63,120 +133,1981 @@ ActorInit En_Kaizoku_InitVars = {
+                ABS_ALT(sp2A) < 0x5000) {
+                if (this->unk_2B0 != 9) {
+                    func_80B87F70(this);
+                    return 1;
+                }
+            } else if (this->unk_2B0 != 3) {
+                func_80B88CD8(this);
+            }
+        }
+        return true;
+    }
+
+    return false;
+}
+
+void EnKaizoku_ChangeAnim(EnKaizoku* this, EnKaizokuAnimation animIndex) {
+    this->animIndex = animIndex;
+    this->frameCount = Animation_GetLastFrame(sAnimations[this->animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->frameCount,
+                     sAnimationModes[this->animIndex], 0.0f);
+}
+
+s32 EnKaizoku_ValidatePictograph(PlayState* play, Actor* actor) {
+    s32 ret = Snap_ValidatePictograph(play, actor, PICTOGRAPH_PIRATE_GOOD, &actor->focus.pos, &actor->shape.rot, 10.0f,
+                                      400.0f, -1);
+
+    ret |= Snap_ValidatePictograph(play, actor, PICTOGRAPH_PIRATE_TOO_FAR, &actor->focus.pos, &actor->shape.rot, 10.0f,
+                                   1200.0f, -1);
+    return ret;
+}
+
+void func_80B85F48(EnKaizoku* this) {
+    Math_Vec3f_Copy(&this->unk_2F8, &gZeroVec3f);
+    Math_Vec3f_Copy(&this->unk_304, &gZeroVec3f);
+    this->unk_2D8 = 1;
+    this->unk_2B0 = 0;
+    this->actionFunc = func_80B85FA8;
+}
+
+void func_80B85FA8(EnKaizoku* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    f32 curFrame = this->skelAnime.curFrame;
+    s32 sp54;
+
+    switch (this->unk_59C) {
+        case 0:
+            if (!(this->picto.actor.xzDistToPlayer < 200.0f)) {
+                break;
+            }
+
+            if (!ActorCutscene_GetCanPlayNext(this->unk_2D6)) {
+                ActorCutscene_SetIntentToPlay(this->unk_2D6);
+                return;
+            }
+
+            ActorCutscene_StartAndSetUnkLinkFields(this->unk_2D6, &this->picto.actor);
+            func_800B7298(play, &this->picto.actor, 0x15);
+            this->subCamId = ActorCutscene_GetCurrentSubCamId(this->picto.actor.cutscene);
+            this->picto.actor.shape.rot.y = this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
+
+            sp54 = (this->unk_2CA * 4) + this->unk_2C8;
+            if (this->unk_2EC != 2) {
+                player->actor.world.pos.x = this->picto.actor.home.pos.x + 90.0f;
+                player->actor.world.pos.z = this->picto.actor.home.pos.z + 30.0f;
+            } else {
+                player->actor.world.pos.x = this->picto.actor.home.pos.x - 90.0f;
+                player->actor.world.pos.z = this->picto.actor.home.pos.z - 30.0f;
+            }
+
+            player->actor.speedXZ = 0.0f;
+            this->picto.actor.world.pos.x = this->picto.actor.home.pos.x;
+            this->picto.actor.world.pos.z = this->picto.actor.home.pos.z;
+            Message_StartTextbox(play, D_80B8A8D0[sp54], &this->picto.actor);
+            this->unk_2C8 += 1;
```

Comment:
```suggestion
            this->unk_2C8++;
```
A few other similar down below.